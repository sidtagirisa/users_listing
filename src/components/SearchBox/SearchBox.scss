@import "../../styles/variables.scss";

.search {
  outline: none;
  line-height: 30px;
  padding: 10px;
  text-align: center;
  width: -webkit-fill-available;
  color: $theme1;
}

.sticky {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  max-width: $max-width;
  margin: auto;
  margin-top: 2%;
  width: 50%;
}