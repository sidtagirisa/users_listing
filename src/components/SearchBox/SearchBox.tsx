import React from 'react';
import './SearchBox.scss';

function SearchBox(props: { placeholder: string; handleChange: any }) {
  return (
    <div className="sticky">
      <input
        className="search"
        type="search"
        placeholder={props.placeholder}
        onChange={props.handleChange}
      />
    </div>
  );
}

export default SearchBox;
