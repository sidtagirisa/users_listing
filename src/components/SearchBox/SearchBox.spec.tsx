import { render } from '@testing-library/react';
import React from 'react';
import SearchBox from '.';

describe('# SearchBox', () => {
  it('should render search box', async () => {
    const { container } = render(
      <SearchBox placeholder="search..." handleChange={() => {}} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
