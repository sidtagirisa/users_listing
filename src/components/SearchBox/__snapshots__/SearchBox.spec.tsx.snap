// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# SearchBox should render search box 1`] = `
<div
  class="sticky"
>
  <input
    class="search"
    placeholder="search..."
    type="search"
  />
</div>
`;
